### PR TITLE
feat: add hardhat-defender package and make hardhat-upgrades global

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
This PR adds the `hardhat-defender` package which is a plugin necessary for use with OpenZeppelin Defender.

I have also decided to move both `hardhat-upgrades` and `hardhat-defender` to the parent package file for them to be available across all smart contract projects.

I only added config settings to the `token` project since it is not used for the `core` project for now and it cannot be moved to the shared Hardhat config since it is a plugin and not a native Hardhat configuration.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,6 +59,8 @@
     "ethers": "5.4.7",
     "@nomiclabs/hardhat-ethers": "yarn:hardhat-deploy-ethers@0.3.0-beta.13",
     "@nomiclabs/hardhat-etherscan": "3.1.0",
+    "@openzeppelin/hardhat-upgrades": "1.19.0",
+    "@openzeppelin/hardhat-defender":  "1.7.0",
```
Comment (XavierEkkel) [inline package.json:63]: nit:
```suggestion
    "@openzeppelin/hardhat-defender": "1.7.0",
```

File: apps/smart-contracts/token/hardhat.config.ts [hunk 2]
```diff
@@ -80,6 +81,10 @@ const config: HardhatUserConfig = {
     disambiguatePaths: false,
     runOnCompile: false,
   },
+  defender: {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/hardhat.config.ts:84]: let's just add this to the `core` dir as well now

File: apps/smart-contracts/token/hardhat.config.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import '@nomiclabs/hardhat-etherscan'
 import '@nomicfoundation/hardhat-chai-matchers'
 import '@typechain/ethers-v5'
 import '@typechain/hardhat'
+import '@openzeppelin/hardhat-defender'
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/hardhat.config.ts:6]: same here

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

**Comment by r0x4xn** [comment]
App <strong>react-boilerplate</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://react-boilerplate-fzdzbms9v-prepo.vercel.app'>https://react-boilerplate-fzdzbms9v-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_3Lydru5gdN5RuqWFQBT8AZrbiROq"></hidden>

**Comment by r0x4xn** [comment]
App <strong>core-dapp</strong> has been deployed to Vercel.

<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://core-dapp-huujyki2x-prepo.vercel.app'>https://core-dapp-huujyki2x-prepo.vercel.app</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="prj_JWVJ3WoV71UcOmB52qrcezjyBL3V"></hidden>

**Pull request merged**